src/main/java/fi/otavanopisto/kuntaapi/server/integrations/ptv/AbstractPtvProvider.java

```diff
@@ -52,5 +84,565 @@
+    result.setNames(translateLocalizedItems(ptvWebPageChannel.getNames()));
+    result.setDescriptions(translateLocalizedItems(ptvWebPageChannel.getDescriptions()));
+    result.setUrls(translateLanguageItems(ptvWebPageChannel.getUrls()));
+    result.setAttachments(translateAttachments(ptvWebPageChannel.getAttachments()));
+    result.setSupportContacts(translateSupportContacts(ptvWebPageChannel.getSupportContacts()));
+    result.setLanguages(ptvWebPageChannel.getLanguages());
+    result.setWebPages(translateWebPages(ptvWebPageChannel.getWebPages()));
+    result.setServiceHours(translateServiceHours(ptvWebPageChannel.getServiceHours()));
+    result.setPublishingStatus(ptvWebPageChannel.getPublishingStatus());
+    
+    return result;
+  }
+
+  protected List<WebPage> translateWebPages(List<fi.otavanopisto.restfulptv.client.model.WebPage> ptvWebPages) {
+    if (ptvWebPages == null) {
+      return Collections.emptyList();
+    }
+
+    List<WebPage> result = new ArrayList<>(ptvWebPages.size());
+
+    for (fi.otavanopisto.restfulptv.client.model.WebPage ptvWebPage : ptvWebPages) {
+      WebPage webPage = translateWebPage(ptvWebPage);
+      if (webPage != null) {
+        result.add(webPage);
+      }
+    }
+
+    return result;
+  }
+  
+  protected WebPage translateWebPage(fi.otavanopisto.restfulptv.client.model.WebPage ptvWebPage) {
+    if (ptvWebPage == null) {
+      return null;
+    }
+
+    WebPage webPage = new WebPage();
+    webPage.setLanguage(ptvWebPage.getLanguage());
+    webPage.setType(ptvWebPage.getType());
+    webPage.setUrl(ptvWebPage.getUrl());
+    webPage.setValue(ptvWebPage.getValue());
+    webPage.setDescription(ptvWebPage.getDescription());
+    
+    return webPage;
+  }
+
+  private OrganizationId translateOrganizationId(String ptvOrganizationId) {
+    OrganizationId organizationPtvId = new OrganizationId(PtvConsts.IDENTIFIFER_NAME, ptvOrganizationId);
+    OrganizationId organizationKuntaApiId = idController.translateOrganizationId(organizationPtvId, KuntaApiConsts.IDENTIFIER_NAME);
+    if (organizationKuntaApiId == null) {
+      logger.severe(String.format("Could not translate organization id %s into Kunta API id", organizationPtvId.getId()));
+      return null;
+    }
+    
+    return organizationKuntaApiId;
+  }
+
+  private List<Address> translateAddresses(List<fi.otavanopisto.restfulptv.client.model.Address> ptvAddresses) {
+    if (ptvAddresses == null) {
+      return Collections.emptyList();
+    }
+
+    List<Address> result = new ArrayList<>(ptvAddresses.size());
+
+    for (fi.otavanopisto.restfulptv.client.model.Address ptvAddress : ptvAddresses) {
+      Address address = translateAddress(ptvAddress);
+      if (address != null) {
+        result.add(address);
+      }
+    }
+
+    return result;
+  }
+  
+  private List<ServiceHour> translateServiceHours(List<fi.otavanopisto.restfulptv.client.model.ServiceHour> ptvServiceHours) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "translateServiceHours" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)